src/Operation/UpdateSearchIndex.php

```diff
@@ -65,7 +66,7 @@ public function __construct(private string $databaseName, private string $collec
      * @throws UnsupportedException if write concern is used and unsupported
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
-    public function execute(Server $server): void
+    public function execute(Server $server): Cursor
```

Comment:
I decided to return the `Cursor` instance returned from `executeCommand` here. We could also return `null` as was done in `DropSearchIndex` and change the return type in an upcoming minor version.

Reply:
We discussed returning `void` for all non-meaningful results: [PHPLIB-1227](https://jira.mongodb.org/browse/PHPLIB-1227)

Reply:
This won't work - best we can do is to return `null` as we can't change a return type to `void` if the interface specifies the method as returning a value.

Reply:
This interface is useless. It is never used to type, and it does not indicate a usable the return type. At best, it guarantees a certain consistency between classes.

This interface is useless. It is never used to type, and in any case it is not capable of indicating the return type. At best, it guarantees a certain consistency between classes.

At the least, I would remove the return type, but I'd be more inclined to remove this interface.

Reply:
Saw this after replying in https://github.com/mongodb/mongo-php-library/pull/1391#discussion_r1759155384.

Removing the interface's return type hint is an easy solution. I also have no objections to removing the interface entirely, as it's internal and doesn't provide any real value. Operations are only ever used interchangeably via the Explainable interface (which really doesn't need to extend Executable).

Reply:
This was changed back to `void`.